### PR TITLE
[Doc][Ray Job] Add `delete` to SDK and CLI doc

### LINKED ISSUE #42275: [Doc] [Ray Job] `delete` is missing from SDK and CLI doc
### Description

The `delete` method was added in https://github.com/ray-project/ray/pull/30056 but it's missing from the following doc pages:
- https://docs.ray.io/en/latest/cluster/running-applications/job-submission/cli.html
- https://docs.ray.io/en/latest/cluster/running-applications/job-submission/jobs-package-ref.html

It should be added to these pages.

### Link

_No response_

### PR DESCRIPTION
**Pull request by chrislevn**
## Why are these changes needed?
From [architkulkarni](https://github.com/architkulkarni)
The `delete` method was added in https://github.com/ray-project/ray/pull/30056 but it's missing from the following doc pages:

https://docs.ray.io/en/latest/cluster/running-applications/job-submission/cli.html
https://docs.ray.io/en/latest/cluster/running-applications/job-submission/jobs-package-ref.html

=> I added the `delete` method in SDK and CLI doc. 

### Result: 

CLI:
Screenshot 2024-01-20 at 8 53 50 PM

SDK: 
Screenshot 2024-01-20 at 8 52 45 PM

## Related issue number
Closes #42275 

## Checks

- [x] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [x] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [x] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [x] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [x] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Comment by chrislevn** [comment]
Hi @architkulkarni . I'm new to open-source contributions. I really like Ray and this is my first PR. Can you help me review this PR? Thank you so much.

File: .gitignore [hunk 1]
```diff
@@ -232,3 +232,6 @@ tag-mapping.json
 
 # Temporary files generated by import sorting linter.
 *.isorted
+
+# Temporary files generated by conda.
+.conda/
```
Comment (architkulkarni) [inline .gitignore:237]:
```suggestion
```

**Review by architkulkarni — COMMENTED** [review]
Looks good! Could you please revert the change to `gitignore`? 

The test failure looks like some unrelated build failure, I'll merge the latest master in case it got fixed upstream.

Could you also find the API stability annotation https://docs.ray.io/en/latest/ray-contribute/stability.html for the SDK `delete` function and change it to "stable"? Right now it's showing up as "alpha" if you click into the docs page https://anyscale-ray--42555.com.readthedocs.build/en/42555/cluster/running-applications/job-submission/doc/ray.job_submission.JobSubmissionClient.delete_job.html#ray.job_submission.JobSubmissionClient.delete_job
Screenshot 2024-01-22 at 9 39 16 AM

File: .gitignore (hunk 1 above)
Comment (architkulkarni) [reply .gitignore:237]: Reminder to revert this change @chrislevn

Comment (architkulkarni) [reply .gitignore:237]: Or if you think it's a necessary change in general for Ray developers, you can open a separate PR for it

File: .github/CODEOWNERS [hunk 2]
```diff
@@ -110,16 +111,17 @@
 # ==== Build and CI ====
 
 # Bazel.
-#/BUILD.bazel @ray-project/ray-core @ray-project/ray-ci-reviewers
-#/WORKSPACE @ray-project/ray-core @ray-project/ray-ci-reviewers
-#/bazel/ @ray-project/ray-core @ray-project/ray-ci-reviewers
+#/BUILD.bazel @ray-project/ray-core @ray-project/ray-ci
```
Comment (aslonnie) [inline .github/CODEOWNERS:114]: can we not change these lines?

**Review by aslonnie — COMMENTED** [review]
incorrect merge or rebase? this PR seems to be wanting to revert many merged PRs.

**Pull request closed without merging**